src/averageAges.js

```diff
@@ -37,7 +49,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  // (arr, bolean) => number
+  const femaleArr = people.filter((person) => {
+    const isFemale = person.sex === 'f';
+
+    return !withChildren
+      ? isFemale
+      : isFemale && people.some(child => child.mother === person.name);
```

Comment:
```suggestion
      : people.some(child => child.mother === person.name);
```
чисто логічно якщо це мама то вона автоматом жінка:)

Reply:
Можна посперечатись! В сучасному світі не все так однозначно ))))